Integrate external errors in form

AlexandraBuzila opened the pull request:
Add support for specifying external errors via `additionalErrors:
ajv.ErrorObject[]` prop.

The additional errors are merged with the validation errors and are
supplied to the affected controls. The additional errors are not
affected by the validationMode like the validation errors are and it is
the responsability of the framework user to update the prop when
applicable (e.g. on ValidationMode changes).

Fixes #1926

Contributed on behalf of STMicroelectronics

sdirix commented on packages/core/src/reducers/core.ts:
```diff
@@ -391,8 +415,11 @@ const getErrorsAt = (
+) => (state: JsonFormsCore): ErrorObject[] => {
+  const errors = state.errors ?? [];
+  const additionalErrors = state.additionalErrors ?? [];
+  return errorsAt(instancePath, schema, matchPath)(state.validationMode === 'ValidateAndHide' ? [...additionalErrors] : [...errors, ...additionalErrors]);
```
```suggestion
  return errorsAt(instancePath, schema, matchPath)(state.validationMode === 'ValidateAndHide' ? additionalErrors : [...errors, ...additionalErrors]);
```

sdirix reviewed (changes requested):
Looks good already!

Can you add an example to the example app which adds/removes multiple additional errors on the click of a button?

AlexandraBuzila commented:
Thank you for the review, Stefan! I updated the code and added examples for React (material and vanilla) and vue-vanilla. The contributed example should work with the angular examples as well when they'll support displaying buttons for the contributed actions like the React app (it's currently not the case).

sdirix reviewed (approved):
Great! Thank you!
